R/guide-colorbar.r

```diff
@@ -244,10 +244,28 @@ guide_geom.colorbar <- function(guide, layers, default_mapping) {
   guide_layers <- lapply(layers, function(layer) {
     matched <- matched_aes(layer, guide, default_mapping)
 
-    if (length(matched) > 0 && (isTRUE(is.na(layer$show.legend)) || isTRUE(layer$show.legend))) {
+    if (length(matched) == 0) {
+      # This layer does not use this guide
+      return(NULL)
+    }
+
+    # check if this layer should be included, different behaviour depending on
+    # if show.legend is a logical or a named logical vector
+    if (is_named(layer$show.legend)) {
+      layer$show.legend <- rename_aes(layer$show.legend)
+      show_legend <- layer$show.legend[matched]
+      # we cannot use `isTRUE(is.na(show_legend))` here because
+      # 1. show_legend can be multiple NAs
+      # 2. isTRUE() was not tolerant for a named TRUE
+      show_legend <- show_legend[!is.na(show_legend)]
+      include <- length(show_legend) == 0 || any(show_legend)
```

Comment:
Why does an empty `show_legend` lead to inclusion here?

Reply:
iiuc, `guide_geom()` shows a legend if

1. the layer's mapping contains any of the corresponding aes and
2. the user does not explicitly choose not to show the legend

So, the absence of the corresponding `show.legend` should be translated as inclusion.

Does this answer your question? Sorry, I might not get your point....

Reply:
You are right - I'm probably just tired

Reply:
Thanks. Sorry for making you busy with reviewing pull requests :)